.github/workflows/build-main.yml

```diff
@@ -20,16 +20,13 @@ jobs:
         ]
     name: Build with Java ${{ matrix.java.version }} (OpenJDK)
     steps:
-      - uses: actions/cache@v2
-        with:
-          path: ~/.m2/repository
-          key: maven-java-${{ matrix.java.version }}
       - uses: actions/checkout@v2
       - uses: actions/setup-java@v2
         name: set up jdk ${{matrix.java.version}}
         with:
           distribution: temurin
           java-version: ${{matrix.java.version}}
+          cache: maven
```

Comment:
It only works because we still compile our artifacts for Java 8, right?

Reply:
It caches the dependencies, not the produced artefacts

Reply:
It's actually the same as the old caching, except that the `setup-java` action nows manages the caching one.